### PR TITLE
feat(harborApi): add ability to overwrite apis

### PR DESCRIPTION
**Pull request by jkurz**
* add ability to overwrite default APIs with environment variables

### CODE REVIEW

File: cmd/harborAPI.go [hunk 1]
```diff
@@ -6,20 +6,36 @@ import (
...
+var catalogitURI = "http://catalogit"
 var triggerURI = "http://harbor-trigger.services.dmtio.net"
 var authAPI = "http://auth.services.dmtio.net"
 var helmitURI = "http://helmit.services.dmtio.net"
-var harborURI = "http://harbor.services.dmtio.net"
-var catalogitURI = "http://catalogit.services.dmtio.net"
+var harborURI = "http://mss-harbor-app"
 var customsURI = "https://customs.services.dmtio.net"
 
+// SetHarborValues will set the default values
```
Comment (jritsema) [inline cmd/harborAPI.go]: comment is out of date

File: cmd/harborAPI.go [hunk 2]
```diff
@@ -6,20 +6,36 @@ import (
...
 	"strings"
 
 	"github.com/jtacoma/uritemplates"
 	"github.com/parnurzeal/gorequest"
 )
 
-var shipItURI = "http://shipit.services.dmtio.net"
+var shipItURI = "http://shipit"
+var catalogitURI = "http://catalogit"
 var triggerURI = "http://harbor-trigger.services.dmtio.net"
```
Comment (jritsema) [inline cmd/harborAPI.go]: Do the other endpoints use a naming convention?

**Review by jritsema — CHANGES_REQUESTED** [review]
Looks good.  Can you add a couple tests that verify the resulting endpoints?

File: cmd/harborAPI.go (hunk 2 above)
Comment (jkurz) [reply cmd/harborAPI.go]: Yes. actually everything uses the same convention.

```
http://${PRODUCT}.${ENVIRONMENT}.services.ec2.dmtio.net
```

**Review by jritsema — DISMISSED** [review]
LGTM

**Review by wilsonodk — COMMENTED** [review]
I really like the idea of being able to override the API addresses. Not a fan of the way the URLs are being concatenated together. I makes big assumptions on the path construction based on a singular value.

**Comment by jtolsma** [comment]
+1 to being able to override API addresses... but magic decisions on 'environment' will just open up pain later if not all environments match.  Better to be explicit IMHO

**Comment by jkurz** [comment]
Updated this PR to allow for overwrites via a config file. Now you can set a file in any directory and specify what urls to hit when using Harbor Compose. Example of doing this is in the CONTRIBUTING.md also in the PR.

File: cmd/harborAPI_test.go [hunk 3]
```diff
@@ -33,3 +38,51 @@ func TestGetShipmentEnvironment(t *testing.T) {
 	assert.Nil(t, err)
 	assert.True(t, success)
 }
+
+func TestReadHarborEndpoints(t *testing.T) {
```
Comment (jritsema) [inline cmd/harborAPI_test.go]: Tests should go in the test file (harborAPI_test.go).

File: cmd/harborAPI.go [hunk 4]
```diff
@@ -4,29 +4,121 @@ import (
...
+	ShipitURI    string `json:"shipit"`
+	CatalogitURI string `json:"catalogit"`
+	TriggerURI   string `json:"trigger"`
+	AuthURI      string `json:"authn"`
+	HelmitURI    string `json:"helmit"`
+	HarborURI    string `json:"harbor"`
+	CustomsURI   string `json:"customs"`
+}
+
+func readConfig() (*Config, error) {
```
Comment (jritsema) [inline cmd/harborAPI.go]: The config management stuff feels maybe a bit much to have in this API client file.  Should we consider moving it to a different file, something like `config.go`?  That might also make more sense should we add non-api related config things in the future.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -20,7 +38,7 @@ Thanks for your interest in the project!  We want to welcome contributors so we
 * Make commits of logical units.
 * Run `go fmt ./...` before committing.
```
Comment (jritsema) [inline CONTRIBUTING.md]: should probably be `go fmt ./cmd`

**Review by jritsema — CHANGES_REQUESTED** [review]

**Comment by jritsema** [comment]
A test failed in the [last build](https://circleci.com/gh/turnerlabs/harbor-compose/309)...need to look at why the build still passed.

File: cmd/harborAPI.go (hunk 4 above)
Comment (jkurz) [reply cmd/harborAPI.go]: yeah I agree.

**Comment by jritsema** [comment]
LGTM

**Review by jritsema — APPROVED** [review]

**Pull request merged**